### PR TITLE
SCons: Move `swappy` option to android and slightly refactor `detect.py`

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by dustdfg** [comment]
Note: moving `swappy` option will be much more useful if https://github.com/godotengine/godot/pull/99055 will be merged

File: platform/android/detect.py [hunk 1]
```diff
@@ -31,6 +31,7 @@ def get_opts():
         ),
         BoolVariable("store_release", "Editor build for Google Play Store (for official builds only)", False),
         BoolVariable("generate_apk", "Generate an APK/AAB after building Android library by calling Gradle", False),
+        BoolVariable("swappy", "Use Swappy Frame Pacing Library", False),
```
Comment (akien-mga) [inline platform/android/detect.py:34]:
```suggestion
        BoolVariable("swappy", "Use Swappy Frame Pacing library", False),
```

**Review by akien-mga — APPROVED** [review]

File: platform/android/detect.py [hunk 2]
```diff
@@ -200,27 +201,27 @@ def configure(env: "SConsEnvironment"):
...
-            env.Append(LIBPATH=["../../thirdparty/swappy-frame-pacing/arm64-v8a"])
+            env.Append(LIBPATH=["#thirdparty/swappy-frame-pacing/arm64-v8a"])
 
     env.Append(CCFLAGS=["-ffp-contract=off"])
 
     # Link flags
 
-    env.Append(LINKFLAGS="-Wl,--gc-sections -Wl,--no-undefined -Wl,-z,now".split())
+    env.Append(LINKFLAGS=["-Wl,--gc-sections", "-Wl,--no-undefined", "-Wl,-z,now"])
     env.Append(LINKFLAGS="-Wl,-soname,libgodot_android.so")
```
Comment (akien-mga) [inline platform/android/detect.py:225]: While you're refactoring these:
```suggestion
    env.Append(LINKFLAGS=["-Wl,-soname,libgodot_android.so"])
```

**Comment by dustdfg** [comment]
Akien's suggestions + rebase

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**